### DISCUSSION TITLE
"Use kits and variants" writes boolean value into settings.json

### DISCUSSION THREAD

**Discussion by monkeber** [microsoft/vscode-cmake-tools#2432] [Q&A] [answered]
Not sure if I should open an issue ticket for this or it's just my setup is somehow wrong.
The problem I'm facing is that when I open a C++ project with CMakePresets.json file which is of the version 1 - I get a VSCode notification about not supporting this version. I click "Use kits and variants" button and it writes a boolean value into my workspace's settings.json for `cmake.useCMakePresets` setting which is, I believe, is incorrect, since it expects string value never/always/auto. I get this notification every time I open VSCode in this project, could it be that some check inside the extension still checks for boolean and gives this warning? Or could it be an error in my setup?

My platform: Kubuntu 20.04
VSCode version: 1.65.2
CMake Tools version: tried 1.10.3 and 1.9.2, both give this notification.

![2022-03-16_13-24](https://user-images.githubusercontent.com/27787571/158579695-67e901ba-855a-4baf-9339-ff28359db00a.png)

**Comment by bobbrow** [comment, marked as answer]
This is a bug.  It should set the value to "never."  Thanks for reporting it.
